lib/webdriveragent.js

```diff
@@ -190,21 +190,48 @@ class WebDriverAgent {
    *   }
    * }
    *
-   * @return {Promise<any?>} State Object
+   * @param {number} [timeoutMs=0] If the given timeoutMs is zero or negative number,
+   *                               this function will return the response of `/status` immediately. If the given timeoutMs,
+   *                               this function will try to get the response of `/status` up to the timeoutMs.
+   * @return {Promise<import('@appium/types').StringRecord|null>} State Object
+   * @throws {Error} If there was an error within timeoutMs timeout.
+   *                 No error is raised if zero or negative number for the timeoutMs.
    */
-  async getStatus () {
+  async getStatus (timeoutMs = 0) {
     const noSessionProxy = new NoSessionProxy({
       server: this.url.hostname,
       port: this.url.port,
       base: this.basePath,
       timeout: 3000,
     });
+
+    const getStatus = async () => await /** @type import('@appium/types').StringRecord */ (noSessionProxy.command('/status', 'GET'));
```

Comment:
maybe it would be better to rename this internal variable, so it does not clash with the function name to not create extra confusions about possible recursion. For example _getStatus or getStatusInternal